### DISCUSSION TITLE
Not getting callback onNotificationOpened in react-native iOS

### DISCUSSION THREAD

**Discussion by saurabh874** [invertase/react-native-firebase#4303] [Q&A] [answered]
Hi @mikehardy 
as discussed with you in the https://github.com/invertase/react-native-firebase/issues/1296#issuecomment-697366015
I am creating an new ticket for this.

issue is when i am getting any new notification and i click on  onNotificationOpened listener should call but its not getting any callback, even in android it is perfectly working.

this.notificationOpenedListener = firebase.notifications().onNotificationOpened((notificationOpen) => {
        badgeCount = 0;
        const { data } = notificationOpen.notification;
        firebase.notifications().setBadge(0);
        const campaignID = _.get(data, 'campaign_id', '');
        if (campaignID && campaignID !== '' && campaignID !== 'null') {
            this.handleCampaign(data);
        } else {
            this.handleContactDetail(data);
        }
        firebase.notifications().removeDeliveredNotification(notificationOpen.notification._notificationId);
    });

this is my sample code.
react-native- 0.61.4
react-natie-firebase-5.6.0

**Comment by andersonaddo** [comment, marked as answer]
v5 of react native firebase isn't supported anymore.
Time to upgrade!

**Reply by saurabh874** [reply]
@andersonaddo  i agree but if i will upgrade version there is some dependency in other party push-notification-ios
and there is lot of issue thats i downgrade.

**Reply by andersonaddo** [reply]
It's worse for you to stay downgraded then to take the pains to upgrade. 

https://rnfirebase.io/faqs-and-tips#i-need-help-with-anything-regarding--v5-of-react-native-firebase-where-could-i-get-help-with-that
